### DISCUSSION TITLE
Implement track changes

### DISCUSSION THREAD

**Discussion by Shiva804** [facebook/lexical#4707] [Q&A] [answered]
Hi,  I am new to lexical editor and I am trying to implement track changes in lexical editor. But I am not sure how to wrap the character typed by the user.

If the user type a character then it should be wrapped in a span tag with class name inserted.
If the user delete a character then it should be wrapped in a span tag with class name deleted.

I created plugin to create a span element but not sure how to replace the character with the created element in the editor.

Can someone give me suggestion on how to implement this efficiently ?

**Comment by CanRau** [comment]
Have you figured it out? If not the [CharacterLimitPlugin](https://github.com/facebook/lexical/blob/4b4db176bc9a373a33f81f11c0e63dee74a25a20/packages/lexical-react/src/shared/useCharacterLimit.ts#L143) from the playground might help which basically does the same, only wrapping characters (or text) which is longer than allowed and then instead of wrapping deleted it unwraps 

Also interesting idea actually, maybe I'm going to use something similar at least for galleries and such 🤔

**Comment by kgpax** [comment]
Hi @Shiva804 .

Did you have any joy implementing this feature?  I am also looking to develop this same functionality in Lexical, and have struggled to find any examples of this having been implemented.

**Comment by Aemulatio** [comment]
I think Im already done this or same thing. 
I've wrote listener for KEY_DOWN command with bigger priority than lexical gives for that command

e.g. base priority is 0 or 1, my registered listener has 2

**Comment by Ramnath-Karthikesan** [comment, marked as answer]
https://github.com/Ramnath-Karthikesan/lexical-track-changes

I've implemented a basic version of track changes in the Lexical Editor. While this approach has some bugs and efficiency issues , contributions to improve the track changes feature in Lexical would be greatly appreciated.

The concept involves creating two nodes, one for insertion and one for deletion, inspired by the structure of the link node in Lexical. A custom plugin is then developed to listen for specific operations such as KEY_DOWN_COMMAND and KEY_BACKSPACE_COMMAND. Based on the current user selection, the plugin executes the node insertion or deletion operation.

**Reply by Aemulatio** [reply]
Duude that looks awesome, where u've been in February?)) 
I'd like to see how u will implemented format and styles tracking

**Reply by Ramnath-Karthikesan** [reply]
thanks mate

**Reply by mdmahendri** [reply]
@Ramnath-Karthikesan bro is there any specific benefits for you to make new nodes extend ElementNode instead of TextNode?

**Reply by feconroses** [reply]
Thanks @Ramnath-Karthikesan, it's working great!

**Comment by mdmahendri** [comment]
For anyone seeking alternative, I made a package [lexical-review](https://github.com/mdmahendri/lexical-review) which customize`TextNode` in `lexical` and tagging insertion and deletion from original text. You can see the demo in https://mdmahendri.github.io/lexical-review/

**Reply by partoa** [reply]
Is there a way to delete one character at a time?

**Reply by mdmahendri** [reply]
EDIT: you can now choose whether to delete one character or one word

**Reply by fabiogif** [reply]
Is it possible to add ToolbarPlugin that have Word features?

**Reply by CanRau** [reply]
@fabiogif what Word features are you referring to? You can basically implement anything you can imagine with some creativity
